zerver/lib/upload.py

```diff
@@ -73,8 +73,8 @@
 class RealmUploadQuotaError(JsonableError):
     code = ErrorCode.REALM_UPLOAD_QUOTA
 
-def attachment_url_to_path_id(attachment_url: str) -> str:
-    path_id_raw = re.sub(r'[/\-]user[\-_]uploads[/\.-]', '', attachment_url)
+def attachment_path_to_path_id(attachment_path: str) -> str:
+    path_id_raw = re.sub(r'user[\-_]uploads[/\.-]', '', attachment_path)
```

Comment:
Why is this needed? According to `urlparse` or `urlsplit`, the path component of `http://chat.zulip.org/user_uploads/foo` is `/user_uploads/foo`, not `user_uploads/foo`.